src/Microsoft.IdentityModel.Clients.ActiveDirectory/Extensions.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Globalization;
+using System.Text;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory
+{
+    internal static class Extensions
+    {
+        internal static string GetPiiScrubbedDetails(this Exception ex)
+        {
+            string res = null;
+            if (ex != null)
+            {
+                var sb = new StringBuilder();
+
+                sb.Append(string.Format(CultureInfo.CurrentCulture, "Exception type: {0}", ex.GetType()));
+
+                if (ex is AdalException)
+                {
+                    var adalException = (AdalException) ex;
+                    sb.Append(string.Format(CultureInfo.CurrentCulture, ", ErrorCode: {0}", adalException.ErrorCode));
+                }
+
+                if (ex is AdalServiceException)
+                {
+                    var adalServiceException  = (AdalServiceException) ex; 
+                    sb.Append(string.Format(CultureInfo.CurrentCulture, ", StatusCode: {0}", adalServiceException.StatusCode));
+                }
+
```

Comment:
Do you need to do anything about AdalClaimsChallengeException?

Reply:
we do not log claims